packages/flutter_tools/lib/src/version.dart

```diff
@@ -709,8 +717,8 @@ class GitTagVersion {
     String gitRef = 'HEAD'
   }) {
     if (fetchTags) {
-      final String channel = _runGit('git rev-parse --abbrev-ref HEAD', processUtils, workingDirectory);
-      if (channel == 'dev' || channel == 'beta' || channel == 'stable') {
+      final String channel = _runGit('git symbolic-ref --short HEAD', processUtils, workingDirectory);
```

Comment:
for posterity, was `git rev-parse --abbrev-ref HEAD` giving us the wrong string?

Reply:
Also, looks like this is a new git sub-command after an internal refactor. I wonder if all of our users will have it in their git? https://git-scm.com/docs/git-symbolic-ref#_notes

Reply:
Well, it was at least around in 2011, so I'm guessing it should be fine https://stackoverflow.com/questions/4986000/whats-the-recommended-usage-of-a-git-symbolic-reference

Reply:
As far as I can tell what we did before was just wrong and this command has existed for ages.
But it's git.
So who knows.
I certainly cannot say with certainty that this is correct.